pallets/polkapool/src/lib.rs

```diff
@@ -327,11 +345,28 @@ decl_module! {
         pub fn slash_stake(origin, account: T::AccountId) -> DispatchResult {
             let origin = <T as Config>::Origin::from(origin);
             T::GovernanceOrigin::ensure_origin(origin)?;
-            let stake = <StakedUsers<T>>::get(&account);
-            <StakedUsers<T>>::remove(&account);
-            T::Currency::slash(AssetId::POLKADEX,&account,stake.staked_amount);
-            Self::deposit_event(RawEvent::StakeSlashed(account,stake.staked_amount));
+            let mut stake_info: StakeInfo<T> = <StakedUsers<T>>::take(&account);
```

Comment:
The governance can slash all the staked tokens of an account completely if this account is found to be spamming the network.